rotkehlchen/rotkehlchen.py

```diff
@@ -923,6 +946,30 @@ def query_asset_movements(
         movements.sort(key=lambda x: x.timestamp, reverse=True)
         return movements
 
+    @overload
+    def _get_price_oracle_order_from_setting(
+            self,
+            price_oracle_setting: List[str],
+            price_oracle_type: Type[CurrentPriceOracle],
+    ) -> List[CurrentPriceOracle]:
+        ...
+
+    @overload
+    def _get_price_oracle_order_from_setting(
+            self,
+            price_oracle_setting: List[str],
+            price_oracle_type: Type[HistoricalPriceOracle],
+    ) -> List[HistoricalPriceOracle]:
+        ...
+
+    # pylint: disable=unused-argument, no-self-use
+    def _get_price_oracle_order_from_setting(  # type: ignore  # TODO fix this
```

Comment:
Do I need to create a NewType for it and remove Type? I'm going to try it

Reply:
One mistake you have here is the return type

```diff
-    ) -> List[Union[CurrentPriceOracle, HistoricalPriceOracle]]:
+    ) -> Union[List[CurrentPriceOracle], List[HistoricalPriceOracle]]:
```
You can never have a list with a mist of different types. But even with this you get

```
Incompatible return value type (got "List[Union[CurrentPriceOracle, HistoricalPriceOracle]]", expected "Union[List[CurrentPriceOracle], List[HistoricalPriceOracle]]")
```

Which leads me to think that this may be a problem since it can't understand that `price_oracle_type` can only be one of the two types and not both. It can't determine what type it can be so it takes the union which fucks up the typing here.